File: tools/web/goshop/js/service/contentService.js
//服务层
var contentUrl='http://localhost:8999/shopping-content-ms/content';
app.service('contentService',function($http){
	    	
	//读取列表数据绑定到表单中
	this.findAll=	function(){
		return $http.get(contentUrl+'/findAll');		
	}
	//分页 
	this.findPage=function(page,rows){
		return $http.get(contentUrl+'/findPage?page='+page+'&rows='+rows);
	}
	//查询实体
	this.findOne=function(id){
		return $http.get(contentUrl+'/findOne?id='+id);
	}
	//增加 
	this.add=function(entity){
		return  $http.post(contentUrl+'/add',entity );
	}
	//修改 
	this.update=function(entity){
		return  $http.post(contentUrl+'/update',entity );
	}
	//删除
	this.dele=function(ids){
		return $http.get(contentUrl+'/delete?ids='+ids);
	}
	//搜索
	this.search=function(page,rows,searchEntity){
		return $http.post(contentUrl+'/search?page='+page+"&rows="+rows, searchEntity);
	}    	
});

File: shopping-parents/shopping-content-ms/src/main/java/com/online/shopping/service/ContentService.java
package com.online.shopping.service;

import java.util.List;

import com.online.shopping.entity.PageResult;
import com.online.shopping.pojo.TbContent;


/**
 * 服务层接口
 * @author Administrator
 *
 */
public interface ContentService {

	/**
	 * 返回全部列表
	 * @return
	 */
	public List<TbContent> findAll();
	
	
	/**
	 * 返回分页列表
	 * @return
	 */
	public PageResult findPage(int pageNum,int pageSize);
	
	
	/**
	 * 增加
	*/
	public void add(TbContent tbContent);
	
	
	/**
	 * 修改
	 */
	public void update(TbContent tbContent);
	

	/**
	 * 根据ID获取实体
	 * @param id
	 * @return
	 */
	public TbContent findOne(Long id);
	
	
	/**
	 * 批量删除
	 * @param ids
	 */
	public void delete(Long [] ids);

	/**
	 * 根据ID获取图片list
	 * @param categoryId
	 * @return
	 */
	public	List<TbContent> findByCategory(Long categoryId);
}
